### PR TITLE
Bugfix/cor 1769 variants table and graph 2

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard29**
Bugfix for COR-1769

### CODE REVIEW

File: packages/app/src/domain/variants/static-props/get-variant-chart-data.ts [hunk 1]
```diff
@@ -18,31 +18,32 @@ const EMPTY_VALUES = {
...
   }
 
-  const variantsOfConcern = variants.values
-    .filter((variant) => variant.last_value.is_variant_of_concern || variant.last_value.has_historical_significance)
-    .filter((variant) => variant.variant_code !== 'other_graph' && variant.variant_code !== 'other_table')
-    .sort((a, b) => b.last_value.order - a.last_value.order);
+  const sortedVariants = variants.values.sort((a, b) => b.last_value.order - a.last_value.order);
 
-  const firstVariant = variantsOfConcern.shift();
+  const firstVariantInList = sortedVariants.shift();
```
Comment (VWSCoronaDashboard30) [inline packages/app/src/domain/variants/static-props/get-variant-chart-data.ts:32]: If we are changing the logic, is this still needed?

@VWSCoronaDashboard29 Have you tested on what happens if we remove this line?

File: packages/app/src/domain/variants/static-props/get-variant-table-data.ts [hunk 2]
```diff
@@ -31,14 +37,14 @@ export function getVariantTableData(variants: NlVariants | undefined, namedDiffe
...
-        return null;
-      }
-
-      return difference;
+      return difference ?? null;
     }
   }
 
+  function mapVariantToNamedDifference(namedDifferenceVariantCode: string) {
+    return variants?.values.find((x) => x.variant_code === namedDifferenceVariantCode) ?? null;
```
Comment (VWSCoronaDashboard30) [inline packages/app/src/domain/variants/static-props/get-variant-table-data.ts:45]: If you're using the nullable symbol, won't this return null by default if nothing is found?

File: packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx [hunk 3]
```diff
@@ -92,10 +92,7 @@ export const VariantsStackedAreaTile = ({ text, values, variantColors, metadata
 
           const reorderContext = {
             ...context,
-            config: [
-              ...context.config.filter((value) => !hasMetricProperty(value) || filteredValues[value.metricProperty] || hasSelectedMetrics),
-              context.config.find((value) => hasMetricProperty(value) && value.metricProperty === 'other_graph_percentage'),
-            ].filter(isDefined),
+            config: context.config.filter((value) => !hasMetricProperty(value) || filteredValues[value.metricProperty] || hasSelectedMetrics).filter(isDefined),
```
Comment (VWSCoronaDashboard30) [inline packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx:95]: I think we need to keep the property spread notation (...) here

File: packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx [hunk 4]
```diff
@@ -35,9 +35,9 @@ export const VariantsStackedAreaTile = ({ text, values, variantColors, metadata
 
   const { list, toggle, clear } = useList<keyof VariantChartValue>(alwaysEnabled);
 
-  const [seriesConfig, otherConfig, selectOptions] = useSeriesConfig(text, values, variantColors);
```
Comment (VWSCoronaDashboard30) [inline packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx:38]: Do you know what otherConfig does?

File: packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx (hunk 3 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx:95]: Why is that?

File: packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx (hunk 4 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx:38]: Yes, it supplies a different visual style for the `other variants` metric on the variants graph. We've agreed with IA, BE and Design that `other variants` are to be treated as all the rest of the variants. Thus, no special configuration is necessary anymore.

File: packages/app/src/domain/variants/static-props/get-variant-table-data.ts (hunk 2 above)
Comment (VWSCoronaDashboard29) [reply packages/app/src/domain/variants/static-props/get-variant-table-data.ts:45]: It wouldn't. It would return `undefined` instead of `null`

Comment (VWSCoronaDashboard30) [reply packages/app/src/domain/variants/static-props/get-variant-table-data.ts:45]: Fair enough

File: packages/app/src/domain/variants/static-props/get-variant-chart-data.ts (hunk 1 above)
Comment (VWSCoronaDashboard30) [reply packages/app/src/domain/variants/static-props/get-variant-chart-data.ts:32]: I got the idea now, thanks!

File: packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx (hunk 3 above)
Comment (VWSCoronaDashboard30) [reply packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx:95]: You're not interacting with the object directly, but with a clone. This way you exclude the possibility of introducing inconsistencies

Comment (VWSCoronaDashboard29) [reply packages/app/src/domain/variants/variants-stacked-area-tile/variants-stacked-area-tile.tsx]: Updated the code

**Review by VWSCoronaDashboard30 — APPROVED** [review]

**Pull request merged**